unblob/finder.py

```diff
@@ -2,166 +2,202 @@
-                        start_offset=real_offset,
-                        exception=exc,
-                    )
-                    task_result.add_report(error_report)
-                    logger.error(
-                        "Unhandled Exception during chunk calculation",
-                        **error_report.asdict()
-                    )
-                    continue
-
-                # We found some random bytes this handler couldn't parse
-                if chunk is None:
-                    continue
-
-                if chunk.end_offset > file_size:
-                    logger.debug("Chunk overflows file", chunk=chunk, _verbosity=2)
-                    continue
-
-                chunk.handler = handler
-                logger.debug(
-                    "Found valid chunk", chunk=chunk, handler=handler.NAME, _verbosity=2
+                return all_chunks
+            else:
+                logger.error(
+                    "Error scanning for patterns",
+                    priority_level=noformat(priority_level),
+                    error=e,
                 )
-                all_chunks.append(chunk)
 
-        logger.debug("Ended priority level", priority_level=noformat(priority_level))
+        logger.debug(
+            "Ended priority level",
+            priority_level=noformat(priority_level),
+            all_chunks=all_chunks,
+        )
 
     return all_chunks
 
 
 @lru_cache
-def make_yara_rules(handlers: Tuple[Type[Handler], ...]):
-    """Make yara.Rule by concatenating all handlers yara rules and compiling them."""
-    all_yara_rules = "\n".join(
-        _YARA_RULE_TEMPLATE.format(NAME=h.NAME, YARA_RULE=h.YARA_RULE.strip())
-        for h in handlers
-    )
-    logger.debug("Compiled YARA rules", rules=all_yara_rules, _verbosity=3)
-    compiled_rules = yara.compile(source=all_yara_rules, includes=False)
-    return compiled_rules
-
-
-@lru_cache
-def make_handler_map(handler_classes: Tuple[Type[Handler], ...]) -> Dict[str, Handler]:
-    return {h.NAME: h() for h in handler_classes}
-
-
-def search_yara_patterns(
-    yara_rules: yara.Rule, handler_map: Dict[str, Handler], full_path: Path
-) -> List[YaraMatchResult]:
-    """Search with the compiled YARA rules and identify the handler which defined the rule."""
-    # YARA uses a memory mapped file internally when given a path
-    yara_matches: List[yara.Match] = yara_rules.match(full_path.as_posix())
-
-    yara_results = []
-    for match in yara_matches:
-        handler = handler_map[match.rule]
-        yara_res = YaraMatchResult(handler=handler, match=match)
-        yara_results.append(yara_res)
+def build_hyperscan_database(
+    handlers: Tuple[Type[Handler], ...]
+) -> Tuple[hyperscan.Database, Dict]:
+    db = hyperscan.Database(mode=hyperscan.HS_MODE_VECTORED)
+    handler_map = dict()
+
+    pattern_id = 0
+    patterns = []
+    for handler in handlers:
+        h = handler()
+        for pattern in handler.PATTERNS:
```

Comment:
This is a little bit confusing, I would:

1. rename the `handler` variable to `handler_class`
2. rename the `h` variable to `handler`
3. keep the `handler.PATTERNS` as `handler.PATTERNS`. 

At this point, we are going through the handler classes, creating a handler object from it and use that object to deal with stuff. I think with these changes, it would be more easier to understand what is going on.

Reply:
done